error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -675,6 +675,34 @@ Flux<T> after(Mono<T> mono) {
     }
   }
 
+  /** Don't unnecessarily convert {@link Mono} to {@link Flux}. */
+  @SuppressWarnings("VoidMissingNullable")
+  static final class MonoFluxThen<T> {
+    @BeforeTemplate
+    Mono<Void> before(Mono<T> mono) {
+      return mono.flux().then();
+    }
+
+    @AfterTemplate
+    Mono<Void> after(Mono<T> mono) {
+      return mono.then();
+    }
+  }
+
+  /** Don't unnecessarily call {@link Mono#then()} on a {@code Mono<Void>} . */
+  @SuppressWarnings("VoidMissingNullable")
+  static final class MonoVoidThen {
+    @BeforeTemplate
+    Mono<Void> before(Mono<Void> mono) {
+      return mono.then();
+    }
+
+    @AfterTemplate
+    Mono<Void> after(Mono<Void> mono) {
+      return mono;
+    }
```

Comment:
We can combine this rule with `MonoSwitchIfEmptyOfEmptyPublisher` above, as we generally try to have one rule for each "target shape" in an `@AfterTemplate`.